internal/services/dataprotection/data_protection_backup_instance_blob_storage_resource.go

```diff
@@ -202,6 +207,7 @@ func resourceDataProtectionBackupInstanceBlobStorageRead(d *schema.ResourceData,
 			d.Set("storage_account_id", props.DataSourceInfo.ResourceID)
 			d.Set("location", props.DataSourceInfo.ResourceLocation)
 			d.Set("backup_policy_id", props.PolicyInfo.PolicyId)
+			d.Set("protection_state", string(pointer.From(props.CurrentProtectionState)))
```

Comment:
nit: we can use `pointer.FromEnum` here

```suggestion
			d.Set("protection_state", pointer.FromEnum(props.CurrentProtectionState))
```